jbpm-flow-builder/src/main/java/org/jbpm/osgi/flow/compiler/Activator.java

```diff
@@ -35,7 +36,8 @@ public void start(BundleContext bc) throws Exception {
         this.processBuilderReg = bc.registerService( new String[]{ ProcessBuilderFactoryService.class.getName(), Service.class.getName()},
                                                                    new ProcessBuilderFactoryServiceImpl(),
                                                                    new Hashtable() );
-        ProcessBuilderFactory.reInitializeProvider();
+        RuntimeManagerFactory.Factory.reset();
+        ProcessBuilderFactory.reInitializeProvider();        
```

Comment:
Ah, thanks. I see now. However, that brings the next question -- should we unify the approach? It seems that the the end result should be the same (a way to re-try to load the provider class)